catalog/resource_share_test.go

```diff
@@ -0,0 +1,331 @@
+package catalog
+
+import (
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	"github.com/databrickslabs/terraform-provider-databricks/qa"
+)
+
+func TestShareCornerCases(t *testing.T) {
+	qa.ResourceCornerCases(t, ResourceShare())
+}
+
+func TestCreateShare(t *testing.T) {
+	qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.0/unity-catalog/shares",
+				ExpectedRequest: ShareInfo{
+					Name: "a",
+					Objects: []SharedDataObject{
+						{
+							Name:           "main.b",
+							DataObjectType: "TABLE",
+							Comment:        "c",
+						},
+						{
+							Name:           "main.a",
+							DataObjectType: "TABLE",
+							Comment:        "c",
+						},
+					},
+				},
+				Response: RecipientInfo{
+					Name: "a",
+				},
+			},
+			{
+				Method:   "PATCH",
+				Resource: "/api/2.0/unity-catalog/shares/a",
+				ExpectedRequest: ShareUpdates{
+					Updates: []ShareDataChange{
+						{
+							Action: "ADD",
+							DataObject: SharedDataObject{
+								Name:           "main.b",
+								DataObjectType: "TABLE",
+								Comment:        "c",
+							},
+						},
+						{
+							Action: "ADD",
+							DataObject: SharedDataObject{
+								Name:           "main.a",
+								DataObjectType: "TABLE",
+								Comment:        "c",
+							},
+						},
+					},
+				},
+				Response: RecipientInfo{
+					Name: "a",
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/unity-catalog/shares/a?include_shared_data=true",
+				Response: ShareInfo{
+					Name: "a",
+					Objects: []SharedDataObject{
+						{
+							Name:           "main.b",
+							DataObjectType: "TABLE",
+							Comment:        "c",
+						},
+						{
+							Name:           "main.a",
+							DataObjectType: "TABLE",
+							Comment:        "c",
+						},
+					},
+				},
+			},
+		},
+		Resource: ResourceShare(),
+		Create:   true,
+		HCL: `
+		name = "a"
+	    objects {
```

Comment:
please add `tf:"alias:object"`, so that HCL DSL makes more sense.

also increase indentation, as it's hard to read.